kotlin/services/kendra/src/main/kotlin/com/example/kendra/CreateIndexAndDataSourceExample.kt

```diff
@@ -0,0 +1,167 @@
+//snippet-sourcedescription:[CreateIndexAndDataSourceExample.kt demonstrates how to create an Amazon Kendra index and data source.]
+//snippet-keyword:[SDK for Kotlin]
+//snippet-keyword:[Code Sample]
+//snippet-service:[Amazon Kendra]
+//snippet-sourcetype:[full-example]
+//snippet-sourcedate:[03/10/2022]
+//snippet-sourceauthor:[scmacdon - aws]
+
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+package com.example.kendra
+
+// snippet-start:[kendra.kotlin.index.import]
+import aws.sdk.kotlin.services.kendra.KendraClient
+import aws.sdk.kotlin.services.kendra.model.DescribeIndexRequest
+import aws.sdk.kotlin.services.kendra.model.CreateDataSourceRequest
+import aws.sdk.kotlin.services.kendra.model.DataSourceType
+import aws.sdk.kotlin.services.kendra.model.DataSourceConfiguration
+import aws.sdk.kotlin.services.kendra.model.S3DataSourceConfiguration
+import aws.sdk.kotlin.services.kendra.model.DescribeDataSourceRequest
+import aws.sdk.kotlin.services.kendra.model.StartDataSourceSyncJobRequest
+import aws.sdk.kotlin.services.kendra.model.DataSourceStatus
+import aws.sdk.kotlin.services.kendra.model.CreateIndexRequest
+import aws.sdk.kotlin.services.kendra.model.IndexStatus
+import kotlinx.coroutines.delay
+import java.util.concurrent.TimeUnit
+import kotlin.system.exitProcess
+// snippet-end:[kendra.kotlin.index.import]
+
+/**
+To run this Kotlin code example, ensure that you have set up your development environment,
+including your credentials.
+
+For information, see this documentation topic:
+https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
+ */
+
+suspend fun main(args: Array<String>) {
+
+    val usage = """
+        Usage:
+            <indexDescription> <indexName> <indexRoleArn> <dataSourceRoleArn> <dataSourceName> <dataSourceDescription> <s3BucketName>
+
+        Where:
+            indexDescription - A description for the index.
+            indexName - The name for the new index.
+            indexRoleArn - An Identity and Access Management (IAM) role that gives Amazon Kendra permissions to access your Amazon CloudWatch logs and metrics.
+            dataSourceRoleArn - The ARN of am IAM role with permission to access the data source.

```

Comment:
dataSourceRoleArn - The Amazon Resource Name (ARN) of an IAM role with permissions to access the data source.

Reply:
fixed